### PR TITLE
cnf network: fix frrnodestatus test case

### PR DESCRIPTION
**Pull request by gkopels**
Fix for issue https://github.com/openshift-kni/eco-gotests/issues/266

### CODE REVIEW

File: tests/cnf/core/network/metallb/tests/frrk8-tests.go [hunk 1]
```diff
@@ -372,10 +385,20 @@ var _ = Describe("FRR", Ordered, Label(tsparams.LabelFRRTestCases), ContinueOnFa
 					// Get the routes
 					frrNodeState, err := metallb.ListFrrNodeState(APIClient)
 					Expect(err).ToNot(HaveOccurred(), "Failed to verify BGP routes")
-					fmt.Println("frrNodeState[0].Objects.Status.RunningConfig",
-						frrNodeState[0].Object.Status.RunningConfig)
 
-					return frrNodeState[0].Object.Status.RunningConfig
+					var frrWorker1Status *metallb.FrrNodeStateBuilder
```
Comment (evgenLevin) [inline tests/cnf/core/network/metallb/tests/frrk8-tests.go:389]: IMO better to create help func to move this logic, something like:
```
func findNodeByName(frrNodeState []metallb.FrrNodeStateBuilder, nodeName string) *metallb.FrrNodeStateBuilder {
    for _, node := range frrNodeState {
        if node.Definition.Name == nodeName {
            return node
        }
    }
    return nil
}
```
and then it should be something like:
```
 frrWorker1Status := findNodeByName(frrNodeState, cnfWorkerNodeList[1].Definition.Name)
    Expect(frrWorker1Status).ToNot(BeNil(), fmt.Sprintf("Node with name %s not found", cnfWorkerNodeList[1].Definition.Name))
```

Comment (gkopels) [reply tests/cnf/core/network/metallb/tests/frrk8-tests.go]: Updated

File: tests/cnf/core/network/metallb/tests/frrk8-tests.go [hunk 2]
```diff
@@ -354,7 +354,20 @@ var _ = Describe("FRR", Ordered, Label(tsparams.LabelFRRTestCases), ContinueOnFa
 					frrNodeState, err := metallb.ListFrrNodeState(APIClient)
 					Expect(err).ToNot(HaveOccurred(), "Failed to verify BGP routes")
 
-					return frrNodeState[0].Object.Status.RunningConfig
+					// Iterate over all the nodes and append worker-0 if present
```
[the remark this answers is not in the document]
Comment (gkopels) [reply tests/cnf/core/network/metallb/tests/frrk8-tests.go]: removed

File: tests/cnf/core/network/metallb/tests/frrk8-tests.go [hunk 3]
```diff
@@ -354,7 +354,11 @@ var _ = Describe("FRR", Ordered, Label(tsparams.LabelFRRTestCases), ContinueOnFa
 					frrNodeState, err := metallb.ListFrrNodeState(APIClient)
```
Comment (ajaggapa) [inline tests/cnf/core/network/metallb/tests/frrk8-tests.go:354]:
```
metallb.ListFrrNodeState(clients.New(kubeconfigpath), client.ListOptions{
		FieldSelector: fields.SelectorFromSet(fields.Set{"metadata.name": "worker-0"})})
```

We can filter worker-0 here  like this

File: tests/cnf/core/network/metallb/tests/frrk8-tests.go [hunk 4]
```diff
@@ -372,10 +376,12 @@ var _ = Describe("FRR", Ordered, Label(tsparams.LabelFRRTestCases), ContinueOnFa
 					// Get the routes
 					frrNodeState, err := metallb.ListFrrNodeState(APIClient)
```
Comment (ajaggapa) [inline tests/cnf/core/network/metallb/tests/frrk8-tests.go:377]:
```
metallb.ListFrrNodeState(clients.New(kubeconfigpath), client.ListOptions{
		FieldSelector: fields.SelectorFromSet(fields.Set{"metadata.name": "worker-1"})})
```

Same like above for worker-1

**Review by ajaggapa — COMMENTED** [review]
Can be simplified by adding filter in existing command. Please try.

**Review by evgenLevin — APPROVED** [review]
/lgtm

File: tests/cnf/core/network/metallb/tests/frrk8-tests.go [hunk 5]
```diff
@@ -351,10 +353,15 @@ var _ = Describe("FRR", Ordered, Label(tsparams.LabelFRRTestCases), ContinueOnFa
...
-					frrNodeState, err := metallb.ListFrrNodeState(APIClient)
+					frrNodeState, err := metallb.ListFrrNodeState(APIClient, client.ListOptions{
+						FieldSelector: fields.SelectorFromSet(fields.Set{"metadata.name": "worker-0"})})
 					Expect(err).ToNot(HaveOccurred(), "Failed to verify BGP routes")
 
-					return frrNodeState[0].Object.Status.RunningConfig
+					frrWorker0Status := findNodeByName(frrNodeState, cnfWorkerNodeList[0].Definition.Name)
+					Expect(frrWorker0Status).ToNot(BeNil(), fmt.Sprintf("Node with name %s not found",
+						cnfWorkerNodeList[0].Definition.Name))
+
```
Comment (ajaggapa) [inline tests/cnf/core/network/metallb/tests/frrk8-tests.go:363]: Remove. No need of this now

File: tests/cnf/core/network/metallb/tests/frrk8-tests.go [hunk 6]
```diff
@@ -351,10 +353,15 @@ var _ = Describe("FRR", Ordered, Label(tsparams.LabelFRRTestCases), ContinueOnFa
...
+					return frrWorker0Status.Object.Status.RunningConfig
```
Comment (ajaggapa) [inline tests/cnf/core/network/metallb/tests/frrk8-tests.go:364]: `return frrNodeState[0].Object.Status.RunningConfig`

Back to original

File: tests/cnf/core/network/metallb/tests/frrk8-tests.go [hunk 7]
```diff
@@ -370,12 +377,15 @@ var _ = Describe("FRR", Ordered, Label(tsparams.LabelFRRTestCases), ContinueOnFa
...
+						FieldSelector: fields.SelectorFromSet(fields.Set{"metadata.name": "worker-1"})})
 					Expect(err).ToNot(HaveOccurred(), "Failed to verify BGP routes")
-					fmt.Println("frrNodeState[0].Objects.Status.RunningConfig",
-						frrNodeState[0].Object.Status.RunningConfig)
 
-					return frrNodeState[0].Object.Status.RunningConfig
+					frrWorker1Status := findNodeByName(frrNodeState, cnfWorkerNodeList[1].Definition.Name)
+					Expect(frrWorker1Status).ToNot(BeNil(), fmt.Sprintf("Node with name %s not found",
+						cnfWorkerNodeList[1].Definition.Name))
+
```
Comment (ajaggapa) [inline tests/cnf/core/network/metallb/tests/frrk8-tests.go:387]: Remove. No need of this now

File: tests/cnf/core/network/metallb/tests/frrk8-tests.go [hunk 8]
```diff
@@ -370,12 +377,15 @@ var _ = Describe("FRR", Ordered, Label(tsparams.LabelFRRTestCases), ContinueOnFa
...
+					return frrWorker1Status.Object.Status.RunningConfig
```
Comment (ajaggapa) [inline tests/cnf/core/network/metallb/tests/frrk8-tests.go:388]: `return frrNodeState[0].Object.Status.RunningConfig`

Back to original

File: tests/cnf/core/network/metallb/tests/frrk8-tests.go [hunk 9]
```diff
@@ -970,3 +980,13 @@ func verifyBlockedRoutes(frrk8sPods []*pod.Builder, blockedPrefixes string) {
...
+
+func findNodeByName(frrNodeState []*metallb.FrrNodeStateBuilder, nodeName string) *metallb.FrrNodeStateBuilder {
+	for _, node := range frrNodeState {
+		if node.Definition.Name == nodeName {
+			return node
+		}
+	}
+
+	return nil
+}
```
Comment (ajaggapa) [inline tests/cnf/core/network/metallb/tests/frrk8-tests.go:992]: Remove

File: tests/cnf/core/network/metallb/tests/frrk8-tests.go (hunk 6 above)
Comment (gkopels) [reply tests/cnf/core/network/metallb/tests/frrk8-tests.go:364]: changed

File: tests/cnf/core/network/metallb/tests/frrk8-tests.go (hunk 8 above)
Comment (gkopels) [reply tests/cnf/core/network/metallb/tests/frrk8-tests.go:388]: changed

File: tests/cnf/core/network/metallb/tests/frrk8-tests.go (hunk 9 above)
Comment (gkopels) [reply tests/cnf/core/network/metallb/tests/frrk8-tests.go:992]: removed

File: tests/cnf/core/network/metallb/tests/frrk8-tests.go (hunk 7 above)
Comment (gkopels) [reply tests/cnf/core/network/metallb/tests/frrk8-tests.go:387]: removed

File: tests/cnf/core/network/metallb/tests/frrk8-tests.go (hunk 5 above)
Comment (gkopels) [reply tests/cnf/core/network/metallb/tests/frrk8-tests.go:363]: removed

**Review by ajaggapa — APPROVED** [review]
lgtm

**Pull request merged**